crates/gpui/src/platform/linux/platform.rs

```diff
@@ -351,6 +351,19 @@ impl<P: LinuxClient + 'static> Platform for P {
         self.reveal_path(path.to_owned());
     }
 
+    fn open_with_system(&self, path: &Path) {
+        let executor = self.background_executor().clone();
+        let path = path.to_owned();
+        executor
+            .spawn(async move {
+                let _ = std::process::Command::new("xdg-open")
+                    .arg(path)
+                    .spawn()
+                    .expect("Failed to open file with xdg-open");
+            })
+            .detach();
+    }
+
```

Comment:
Don't hard-code shelling out to a program that isn't even listed as a dependency of the crate. The right way to do this is to use a library that understands mime associations (e.g. https://docs.rs/xdg-utils/latest/xdg_utils/). In the worst case the command should be configurable.